posthog/models/person.py

```diff
@@ -50,6 +50,19 @@ def merge_people(self, people_to_merge: List["Person"]):
 
             capture_internal(event, self.distinct_ids[-1], None, None, now, now, self.team.id)
 
+    def split_person(self, main_distinct_id: str):
+        with transaction.atomic():
+            distinct_ids = Person.objects.get(pk=self.pk).distinct_ids
+            if not main_distinct_id:
+                self.properties = {}
+                self.save()
+                main_distinct_id = distinct_ids[0]
+
+            PersonDistinctId.objects.filter(person=self).exclude(distinct_id=main_distinct_id).delete()
+            for distinct_id in distinct_ids:
+                if not distinct_id == main_distinct_id:
+                    Person.objects.create(team_id=self.team_id, distinct_ids=[distinct_id])
```

Comment:
Not 100% if running this in an atomic transaction block will execute this in a single statement in Postgres, but you could optimize by using `bulk_create`, considering there might be persons with a ton of IDs.

Reply:
I did think about this but there are 2 issues
1. Clickhouse post-save hooks [won't be triggered](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#django.db.models.query.QuerySet.bulk_create)
2. The distinct ids are actually created as a separate row and with bulk_create we wouldn't have the person id to create the distinct id mapping.